bundles/org.openhab.binding.twitter/src/main/java/org/openhab/binding/twitter/internal/TwitterHandler.java

```diff
@@ -0,0 +1,298 @@
+     *            the file to attach. May be null if no attached file.
+     *
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    @SuppressWarnings("null")
+    private boolean sendTweet(final String tweetTxt, final @Nullable File fileToAttach) {
+        if (!checkPrerequisites()) {
+            return false;
+        }
+        // abbreviate the Tweet to meet the 280 character limit ...
+        String abbreviatedTweetTxt = StringUtils.abbreviate(tweetTxt, CHARACTER_LIMIT);
+        try {
+            // send the Tweet
+            StatusUpdate status = new StatusUpdate(abbreviatedTweetTxt);
+            if (fileToAttach != null && fileToAttach.isFile()) {
+                status.setMedia(fileToAttach);
+            }
+            Status updatedStatus = client.updateStatus(status);
+            logger.debug("Successfully sent Tweet '{}'", updatedStatus.getText());
+            updateState(CHANNEL_LASTTWEET, StringType.valueOf(updatedStatus.getText()));
+            return true;
+        } catch (TwitterException e) {
+            logger.warn("Failed to send Tweet '{}' because of : {}", abbreviatedTweetTxt, e.getLocalizedMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Sends a standard Tweet.
+     *
+     * @param tweetTxt
+     *            text string to be sent as a Tweet
+     *
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    public boolean sendTweet(String tweetTxt) {
+        if (!checkPrerequisites()) {
+            return false;
+        }
+        return sendTweet(tweetTxt, (File) null);
+    }
+
+    /**
+     * Sends a Tweet with an image
+     *
+     * @param tweetTxt
+     *            text string to be sent as a Tweet
+     * @param tweetPicture
+     *            the path of the picture that needs to be attached (either an url,
+     *            either a path pointing to a local file)
+     *
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    public boolean sendTweet(String tweetTxt, String tweetPicture) {
+        if (!checkPrerequisites()) {
+            return false;
+        }
+
+        // prepare the image attachment
+        File fileToAttach = null;
+        boolean deleteTemporaryFile = false;
+        if (StringUtils.startsWith(tweetPicture, "http://") || StringUtils.startsWith(tweetPicture, "https://")) {
+            // we have a remote url and need to download the remote file to a temporary location
+            String tDir = System.getProperty("java.io.tmpdir");
+            String path = tDir + File.separator + "openhab-twitter-remote_attached_file" + "."
+                    + FilenameUtils.getExtension(tweetPicture);
+            try {
+                URL url = new URL(tweetPicture);
```

Comment:
The OH framework has an integrated HTTP client that should be used to save resources. See https://next.openhab.org/javadoc/latest/org/openhab/core/io/net/http/httputil